src/cc/lib/graph/partition.cc

```diff
@@ -441,9 +441,11 @@ void Partition::GetNodeStringFeature(uint64_t internal_node_id, std::span<const
     }
 }
 
-size_t Partition::FullNeighbor(uint64_t internal_id, std::span<const Type> edge_types,
-                               std::vector<NodeId> &out_neighbors_ids, std::vector<Type> &out_edge_types,
-                               std::vector<float> &out_edge_weights) const
+template <class F>
+size_t NeighborIndexIterator(uint64_t internal_id, std::span<const Type> edge_types, F func,
+                             std::vector<uint64_t> m_neighbors_index, std::vector<Type> m_edge_types,
```

Comment:
Please, don't copy entire graph indices, edge types and type offsets. Pass these vectors by reference instead of value..